tools/RAiDER/cli/__main__.py

```diff
@@ -0,0 +1,455 @@
+
+    add_cpus(misc)
+    add_verbose(misc)
+
+    args =  p.parse_args()
+
+    dlGNSS(args)
+    return
+
+
+## ------------------------------------------------------------ prepFromARIA.py
+def calcDelaysARIA(iargs=None):
+    from RAiDER.aria.prepFromARIA import main as ARIA_prep
+    from RAiDER.aria.calcARIA    import main as ARIA_calc
+
+    p = argparse.ArgumentParser(
+        description='Prepare files from ARIA-tools output to use with RAiDER')
+
+    p.add_argument(
+        'file', type=str,
+        help='1 ARIA GUNW netcdf file'
+        )
+
+    p.add_argument(
+        '-m', '--model', default='ERA5', type=str,
+        help='Weather model (default=HRRR)'
+        )
+
+    p.add_argument(
+        '-d', '--dem', default='', type=str,
+        help='Custom DEM. Default: Copernicus GLO30 (downloaded on fly)'
+        )
+
+    p.add_argument(
+        '-o', '--output_directory', default=os.getcwd(), type=str,
+        help='Directory to store results. Default: ./'
+        )
+
+    p.add_argument(
+        '-w', '--write', default=True,
+        help='Optionally write the delays into the GUNW products'
+        )
+
+    ## always doing ray
+    # p.add_argument(
+    #     '-s', '--slant',  choices=('projection', 'ray'),
+    #     type=str, default='projection',
+    #     help='Delay calculation projecting the zenith to slant or along rays ')
+
+    # p.add_argument(
+    #     '-o', '--orbit_directory', default='./orbits', type=str,
+    #     help='Path where orbits will be downloaded')
+
+    # p.add_argument(
+    #     '--los_convention', '-lc', choices=('isce', 'hyp3'), default='isce',
+    #             type=str, help='LOS convention')
+
+    # p.add_argument(
+    #     '--los_file', default='los.geo', type=str,
+    #     help='Output line-of-sight filename')
+
+    # Line of sight
+    # p.add_argument(
+    #     '--incFile', type=str,
+    #     help='GDAL-readable raster image file of inclination (optional)')
+    #
+    # p.add_argument(
+    #     '--azFile', type=str,
+    #     help='GDAL-readable raster image file of azimuth (optional)')
+
+    # p.add_argument(
+    #     '--format', '-t', default='.tif', type=str, dest='fmt',
+    #     help='Output file format (default=tif)')
+
+    args       = p.parse_args(args=iargs)
+    args.argv  = iargs if iargs else os.sys.argv[1:]
+    # args.files = glob.glob(args.files)
+    args.file  = args.file
+
+    ## prep the files needed for delay calcs
+    path_cfg   = ARIA_prep(args)
+    ## do the delay calcs
+    # call(['raider.py', path_cfg])
+
+    tropoDelayFile = calcDelays([path_cfg])
+
+
+    ## calculate the interferometric phase and write it out
+    # ARIA_calc(tropoDelayFile, args.write)
+
+    return
+
+
+def main():
+    parser = argparse.ArgumentParser(prefix_chars='+',
+                        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '++process', choices=['calcDelays', 'downloadGNSS', 'calcDelaysARIA'],
```

Comment:
See comment on CalcDelelaysARIA. 
Would be good to split the CalcDelays also in a download and load component.
At min it should be easy to split the calcDelay from the pre-process model as its already exposed in the function.

Reply:
@bbuzz31 @dbekaert this is one of the complexities I encountered before when trying to implement the "steps" approach. If you want to break out the download/preprocess/calculate delays steps it's not very straightforward without using something like steps. Perhaps a compromise would be to allow each step as separate workflows.

Reply:
Actually with this CLI it's not too difficult to do. But I think it adds unnecessary complexity. There are still a bunch of parameters the user would need to pass, so a config file is still pretty useful.

I've set it up that calcDelay (the default) can take `--download_only` to just stop processing at the weather model. This can also be specified in the configure file.

In the future, we could easily add another ++process like 'processWM' that could just take a few optional parameter (like model and bbox) and then compute for a few default times etcc...